sessionize/app/src/main/java/co/touchlab/sessionize/feedback/FeedbackDialog.kt

```diff
@@ -0,0 +1,196 @@
+import android.widget.Button
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import co.touchlab.sessionize.AppContext
+import co.touchlab.sessionize.R
+import kotlinx.android.synthetic.main.feedback_view.*
+
+enum class FeedbackRating(val value: Int) {
+    None(0),
+    Good(1),
+    Ok(2),
+    Bad(3)
+}
+
+class FeedbackDialog : DialogFragment(),FeedbackInteractionInterface{
+
+
+    private var ratingView:FeedbackRatingView? = null
+    private var commentView:FeedbackCommentView? = null
+
+    private var doneButton:Button? = null
+
+    private val animationTime = 400L
+
+    private var sessionId:String? = null
+    private var sessionTitle:String? = null
+    private var rating:FeedbackRating = FeedbackRating.None
+    private var comments:String = ""
+
+    private var feedbackManager:FeedbackManager? = null
+
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        return activity?.let {
+
+            val feedbackView = createFeedbackView(requireActivity().layoutInflater)
+
+            val builder = AlertDialog.Builder(it)
+            builder.setView(feedbackView)
+                    .setNegativeButton("Close and Disable Feedback", DialogInterface.OnClickListener { dialog, _ ->
+                        AppContext.disableFeedback()
+                        dialog.dismiss()
+                    })
+
+            val dialog = builder.create()
+            dialog.setOnShowListener {
+                dialog.getButton(DialogInterface.BUTTON_NEGATIVE).setTextColor(Color.GRAY)
+            }
+            return dialog
+        } ?: throw IllegalStateException("Activity cannot be null")
+    }
+
+    private fun createFeedbackView(inflater: LayoutInflater):View{
+        val view = inflater.inflate(R.layout.feedback_view, null)
+        view?.let { fbView ->
+            doneButton = fbView.findViewById(R.id.doneButton)
+            doneButton?.setOnClickListener {
+                finishAndClose()
+            }
+            doneButton?.isEnabled = false
+
+            initRatingView(fbView)
+            initCommentView(fbView)
+        }
+        return view
+    }
+
+    private fun initRatingView(feedbackView:View){
+        ratingView = feedbackView.findViewById(R.id.ratingView)
+        ratingView?.createButtonListeners()
+        ratingView?.setFeedbackInteractionListener(this)
+        ratingView?.createCommentButtonListener(View.OnClickListener {
+            showCommentView()
+        })
+        sessionTitle?.let {
+            ratingView?.setSessionTitle(it)
+        }
+
+    }
+
+    private fun initCommentView(feedbackView:View){
+        commentView = feedbackView.findViewById(R.id.commentView)
+        commentView?.createButtonListeners()
+        commentView?.setFeedbackInteractionListener(this)
+        commentView?.visibility = View.INVISIBLE
+    }
+
+    fun setSessionInfo(sessionId: String,sessionTitle:String){
```

Comment:
moving sessionId, sessionTitle, feedbackManager to the constructor would eliminate the need for these functions and let them be non-nullable

Reply:
I moved it to a constructor, however since it's a Fragment it gave me a warning that we should be using SetArguments. Current I disabled the warning and added a constructor, but maybe we want to use that method instead?

Reply:
hmm, yeah you may want to make a static "newInstance" and set arguments there. Forgot it was a dialog, always a pain. Will want to make sure things are handled correctly with config change

Reply:
I've added a newInstance function in my latest commit.